### PR TITLE
Alert to_json("v6") Unit Tests

### PR DESCRIPTION
**Pull request by kebringer-cb**
Doesn't meet any standards, not for merging, sharing the test file for troubleshooting.

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

* CBAPI-4940 - unit tests for Alert to_json("v6") method. This is the majority of the code
* CBAPI-4968 - change in Alerts.py corrects to a static method.

## Pull Request Description

* Add mocks and unit tests for each of the Alert types supported in SDK 1.4.3 to verify the functionality of SDK 1.5.0 to_json("v6") method.

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## How Has This Been Tested?

Executed locally and manual verification of test data and expected results.

### CODE REVIEW

File: src/tests/unit/platform/test_alertsv7_v6_compatibility.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    Test the generation of a v6 to_json output
+
+    Compare what is generated by the current SDK with expected from SDK 1.4.3
+    Parameterization above is used to call this test multiple times to test different alert types
+    """
+    # set up the mock request and execute the mock v7 API call
+    cbcsdk_mock.mock_request("GET", url, v7_api_response)
+    api = cbcsdk_mock.api
+    alert = api.select(BaseAlert, v6_sdk_response.get("id"))
+    print("\n\nStarting comparison of {} type alerts".format(alert.get("type")))
```
Comment (avanbrunt-cb) [inline src/tests/unit/platform/test_alertsv7_v6_compatibility.py:135]: We don't tend to have print statements in tests to reduce the noise log statements are fine as they get suppressed especially when passing

File: src/tests/unit/platform/test_alertsv7_v6_compatibility.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+    # fields from watchlist alert that are not in v7. No mapping available
+    assert not (
+        alert_type == "WATCHLIST" and key in WATCHLIST_FIELDS_V6 and key in alert_v6_from_v7
+    ), ("ERROR: Field is deprecated and does not exist in v7. Expected: Not in to_json(v6). Actual: was incorrectly "
+        "included. Source: WATCHLIST_FIELDS_V6. Key: {}").format(key)
+
+    # check dict recursively
+    # defensively coded.  This method should only be called with dicts, but avoid crashing
+    assert(isinstance(alert_v6, dict)), "Function check_dict called with incorrect argument types"
```
Comment (avanbrunt-cb) [inline src/tests/unit/platform/test_alertsv7_v6_compatibility.py:198]: This should probably be at the top of the function

Comment (kebringer-cb) [reply src/tests/unit/platform/test_alertsv7_v6_compatibility.py:198]: moved

File: src/tests/unit/platform/test_alertsv7_v6_compatibility.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    """Mocks CBC SDK for unit tests"""
+    return CBCSDKMock(monkeypatch, cb)
+
+
+# ==================================== UNIT TESTS BELOW ====================================
+
+# Fields that are not tested for mappings while bugs are fixed.
+# When all bugs are fixed, SKIP_FIELDS should be empty and all tests should pass
+SKIP_FIELDS = {
+    "comment",  # cbapi-4969, TO DO DEFECT FOR CLASSIFICATION
```
Comment (kebringer-cb) [inline src/tests/unit/platform/test_alertsv7_v6_compatibility.py:51]: reminder - update these tests when related issues are addressed

Comment (kebringer-cb) [reply src/tests/unit/platform/test_alertsv7_v6_compatibility.py]: Done.

File: src/tests/unit/platform/test_alertsv7_v6_compatibility.py (hunk 1 above)
Comment (kebringer-cb) [reply src/tests/unit/platform/test_alertsv7_v6_compatibility.py]: removed

**Review by avanbrunt-cb — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@ repos:
     hooks:
     -   id: flake8
         additional_dependencies: [flake8-docstrings]
-        args: ["--docstring-convention", "google"]
+        args: [ "--extend-ignore", "W503", "--docstring-convention", "google"]
```
[the remark this answers is not in the document]
Comment (avanbrunt-cb) [reply .pre-commit-config.yaml]: The link actually indicates W504 replaced W503 ignoring W503 instead

**Pull request merged**